### PR TITLE
NAS-122113 / 23.10 / Initial ccache support

### PR DESCRIPTION
**Pull request by bmeagherix**
This PR only includes **baseline** (aka initial) `ccache` support for developers who choose to enable it, and further refinement will need to take place as its effectiveness is limited.***

It is currently turned **off** by default, and can be enabled either by editing the _Makefile_, or by providing e.g. `CCACHE_DIR=/root/.ccache` on the `make` command line.  This **must** (also) be done when creating the bootstrapdir, as it will install _ccache_ and modify the `PATH` in _root/.bashrc_ there.

*** Currently this:

- **Does** cache/accelerate kernel re-compilation
- **Does not** cache/accelerate zfs, scst or samba re-compilation - reason TBD

(Since kernel compilation is the long-pole, decided with proceed with this PR as-is, rather than wait until perfect.)

### CODE REVIEW

**Review by sonicaj — COMMENTED** [review]
Taking this on slack next

**Comment by bmeagherix** [comment]
@sonicaj Based upon our conversation I've reverted the commit, as IMO it would just muddy the water when you push the basis of the new approach.  We still have the original commit and can copy out anything that is still useful.

**Comment by sonicaj** [comment]
The build time got reduced by almost half for me:
```
❯ env CCACHE=1 make packages PACKAGES="kernel"
Setting up new virtual environment
. ./venv-f9aee55/bin/activate && scale_build packages --packages kernel
[2023-05-24 12:18:23] System state Validated
[2023-05-24 12:18:23] Manifest Validated
[2023-05-24 12:18:23] Building packages (./logs/build_packages.log)
[2023-05-24 12:18:23] Setting up bootstrap directory
[2023-05-24 12:18:28] Successfully setup bootstrap directory
[2023-05-24 12:18:30] 43 package(s) do not need to be rebuilt (openseachest,licenselib,throttle,truenas,pyglfs,zectl,truenas_webui,rrdcached,swagger,truenas_binaries,python_netsnmpagent,glustercli_python,scst,k3s,py_fenced,py_nvme,middlewared,nfs4xdr_acl_tools,kube_router,parted,zettarepl,collectd,ixdiagnose,plugins,sedutil,chelsio_uwire,openzfs-dbg,truenas_samba,inadyn,avahi,truenas_files,py_libzfs,grub2,py_sgio,midcli,rclone,openzfs,kernel-dbg,catalog_validation,truenas_installer,python_truenas_requirements,multus_cni,scst-dbg)
[2023-05-24 12:18:30] Going to build 1 package(s): kernel
[2023-05-24 12:18:30] Creating 1 parallel task(s)
[2023-05-24 12:18:30] Building 'kernel' package
[2023-05-24 13:07:07] Successfully built 'kernel' package (Remaining 0 packages)
[2023-05-24 13:07:07] Success! Done building packages
❯ env CCACHE=1 make packages PACKAGES="kernel"
. ./venv-f9aee55/bin/activate && scale_build packages --packages kernel
[2023-05-24 13:09:18] System state Validated
[2023-05-24 13:09:18] Manifest Validated
[2023-05-24 13:09:18] Building packages (./logs/build_packages.log)
[2023-05-24 13:09:18] Setting up bootstrap directory
[2023-05-24 13:09:23] Successfully setup bootstrap directory
[2023-05-24 13:09:25] 43 package(s) do not need to be rebuilt (grub2,py_sgio,truenas_binaries,kube_router,zettarepl,midcli,chelsio_uwire,truenas_files,k3s,ixdiagnose,swagger,truenas_webui,rclone,glustercli_python,middlewared,scst-dbg,python_netsnmpagent,multus_cni,openzfs-dbg,collectd,py_nvme,pyglfs,avahi,py_fenced,throttle,openzfs,rrdcached,kernel-dbg,zectl,parted,truenas_installer,scst,catalog_validation,truenas_samba,inadyn,nfs4xdr_acl_tools,licenselib,openseachest,py_libzfs,truenas,sedutil,plugins,python_truenas_requirements)
[2023-05-24 13:09:25] Going to build 1 package(s): kernel
[2023-05-24 13:09:25] Creating 1 parallel task(s)
[2023-05-24 13:09:25] Building 'kernel' package
[2023-05-24 13:36:35] Successfully built 'kernel' package (Remaining 0 packages)
[2023-05-24 13:36:35] Success! Done building packages
```

First run is a clean run, i guess ccache sets up everything and it normally takes the same amount of time to build the package - second run is however quite interesting and we can notice the performance gains.

File: conf/build.manifest
Comment (yocalebo) [inline conf/build.manifest:242]: As discussed in slack, let's rename this to `supports_ccache`.

**Review by yocalebo — APPROVED** [review]
This looks good to me, however, please wait for @bmeagherix final review since he's much more familiar with the process.

**Comment by bmeagherix** [comment]
Looks great to me!

(And much cleaner/better than my original attempt, so thanks!)

**Pull request merged**
